spec/javascripts/models/file_uploader_spec.js

```diff
@@ -1,103 +1,209 @@
+      it('emits new:batch event on first add after submit', function() {
+        var fileUploader = new pageflow.FileUploader({
+          entry: this.entry,
+          fileTypes: this.fileTypes
+        });
+        var upload = {name: 'image.png', type: 'image/png'};
+        var handler = sinon.spy();
+
+        fileUploader.on('new:batch', handler);
+        fileUploader.add(upload);
+        fileUploader.submit();
+        fileUploader.add(upload);
+
+        expect(handler).to.have.been.calledTwice;
+      });
 
-      expect(handler).to.have.been.calledTwice;
+      it('throws exception if target set', function() {
+        var fileUploader = new pageflow.FileUploader({
+          entry: this.entry,
+          fileTypes: this.fileTypes
+        });
+        var targetFileUpload = {name: 'video.mp4', type: 'video/mp4'};
+        pageflow.config = {confirmEncodingJobs: false};
+        fileUploader.add(targetFileUpload);
+        fileUploader.submit();
+        var targetFile = this.entry.getFileCollection(this.videoFileType).first();
+        var nonNestedUpload = {name: 'nested_video.mp4', type: 'video/mp4'};
+        var editor = new pageflow.EditorApi();
+        editor.setUploadTargetFile(targetFile);
+
+        expect(function() {
+          fileUploader.add(nonNestedUpload, {editor: editor});
+        }).to.throw(pageflow.InvalidNestedTypeError);
+      });
     });
 
-    it('emits new:batch event on first add after submit', function() {
-      var fileUploader = new pageflow.FileUploader({
-        entry: this.entry,
-        fileTypes: this.fileTypes
+    describe('nested file', function() {
+      it('adds file to nested files collection of file type on target file', function() {
+        var fileUploader = new pageflow.FileUploader({
+          entry: this.entry,
+          fileTypes: this.fileTypes
+        });
+        var targetFileUpload = {name: 'video.mp4', type: 'video/mp4'};
+        pageflow.config = {confirmEncodingJobs: false};
+        fileUploader.add(targetFileUpload);
+        fileUploader.submit();
+        var targetFile = this.entry.getFileCollection(this.videoFileType).first();
+        var nestedFileUpload = {name: 'text_track.vtt', type: 'text/vtt'};
+        var editor = new pageflow.EditorApi();
+        editor.setUploadTargetFile(targetFile);
+        fileUploader.add(nestedFileUpload, {editor: editor});
+        sinon.stub(this.entry, 'getFileCollection').withArgs(this.textTrackFileType)
+          .returns(new pageflow.FilesCollection.createForFileType(this.textTrackFileType,
+                                                                  {},
+                                                                  {entry: this.entry,
+                                                                   parentFile: targetFile}));
+        expect(editor.nextUploadTargetFile.nestedFiles(
+          this.entry.getFileCollection(this.textTrackFileType)
+        ).length).to.eq(1);
```

Comment:
What does this stub do? The only one calling `this.entry.getFileCollection` after it has been stubbed, is the test itself, to get the parameter for the `nestedFiles` call. If we want to pass in a newly created collection, we can just do that directly. Moreover, far as I can see, the collection returned by the stub is empty (we pass `{}` as `files` paramter, which I guess should be `[]`). So I'm not sure why the test is passing.